### ISSUE TITLE
Add namespace into Pod metadata.namespace when ReplicaSet/Job/Daemonset controller create Pods

### ISSUE THREAD

**Issue by k2huang** [kubernetes/kubernetes#89345]
**What would you like to be added**:
Add namespace into Pod metadata.namespace when ReplicaSet/Job/Daemonset controller create Pods

**Why is this needed**:
In my [MutatingAdmissionWebhook](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#mutatingadmissionwebhook) server, I will create one PVC object and append it into Pod.Spec.Volumes after receiving Pod Creation Event, which need get Pod namespace at first. But when using Deployment workload to create Pods, I found I can't get Pod's namespace from Pod Object in my MutatingAdmissionWebhook server, and StatefulSet workload is ok.

Then I read the code of Deployment and StatefulSet controllers, and find: StatefulSet Controller will [set StatefulSet Object's namespace as Pod's namespace when creating Pod](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/statefulset/stateful_set_utils.go#L195), But Deployment/ReplicaSet/Job/Daemonset Controller [don't do this like StatefulSet](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/controller_utils.go#L525)。Can we add it in [this function](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/controller_utils.go#L569)?

**Comment by liggitt** [comment]
xref https://github.com/kubernetes/kubernetes/issues/88282

the namespace should be read from the admission attributes in create requests

**Comment by k2huang** [comment]
> xref #88282
> 
> the namespace should be read from the admission attributes in create requests

Yes, we can get it from admission attributes. But I'm confused why ReplicaSet/Job/Daemonset controllers don't add namespace when creating Pod as StatefulSet Controller.
